apps/src/fish/FishView.jsx

```diff
@@ -53,23 +47,98 @@ class FishView extends React.Component {
     mobilePortraitWidth: PropTypes.number.isRequired
   };
 
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      appWidth: $('#codeApp').width(),
+      appHeight: $('#codeApp').height()
+    };
+  }
+
   componentDidMount() {
     this.props.onMount();
+
+    this.setState({
+      windowWidth: $(window).width(),
+      windowHeight: $(window).height(),
+      appWidth: $('#codeApp').width(),
+      appHeight: $('#codeApp').height()
+    });
+
+    window.addEventListener('resize', _.debounce(this.onResize, 100));
   }
 
+  onResize = () => {
+    const windowWidth = $(window).width();
+    const windowHeight = $(window).height();
+
+    // Check that the window dimensions have actually changed to avoid
+    // unnecessary event-proessing on iOS Safari.
+    if (
+      this.state.windowWidth !== windowWidth ||
+      this.state.windowHeight !== windowHeight
+    ) {
+      const appWidth = $('#codeApp').width();
+      const appHeight = $('#codeApp').height();
+
+      this.setState({windowWidth, windowHeight, appWidth, appHeight});
+    }
+  };
+
   render() {
+    // The tutorial has a width:height ratio of 16:9.
+    const aspectRatio = 16 / 9;
+
+    let containerWidth, containerHeight;
+
+    // Constrain tutorial to 1280px maximum width.
+    const maxContainerWidth = Math.min(this.state.appWidth, 1280);
+
+    // Reducing by 27px leaves appropriate space above the small footer.
+    const maxContainerHeight =
+      Math.min(this.state.appHeight, window.innerHeight) - 27;
```

Comment:
I'd create constants for the aspectRatio, the 1280 value, and the 27 value.

And the 320, 18, and 930 below!

The explicit call to `window.innerHeight` is interesting - shouldn't the same value be available in `state.windowWidth` (unless the "inner" part is important)